model_analyzer/config/run/run_config_generator.py

```diff
@@ -13,181 +13,69 @@
     exection of a list of run configurations.
     """
-
-    def __init__(self,
-                 model,
-                 analyzer_config,
-                 client,
-                 server,
-                 result_manager,
-                 metrics_manager,
-                 run_search=None,
-                 generate_only=False):
+    def __init__(self, config, client):
         """
-        analyzer_config : ModelAnalyzerConfig
+        Parameters
+        ----------
+        config : ModelAnalyzerConfig
             The config object parsed from
             a model analyzer config yaml or json.
-
-        model : ConfigModel
-            ConfigModel object to generate the run configs for.
-
         client : TritonClient
             TritonClient to be used for interacting with the Triton API
         """
 
-        self._analyzer_config = analyzer_config.get_all_config()
-        self._model = model
+        self._analyzer_config = config.get_all_config()
         self._run_configs = []
-        self._run_search = run_search
         self._client = client
-        self._server = server
-        self._run_search = run_search
-        self._metrics_manager = metrics_manager
-        self._result_manager = result_manager
         self._model_name_index = 0
         self._model_configs = []
-        self._generate_only = generate_only
-        self._generate_run_configs()
 
-    def _generate_model_config_combinations(self, value):
+    def run_configs(self):
         """
-        Generates all the alternative config fields for
-        a given value.
-
-        Parameters
-        ----------
-        value : object
-            The value to be used for sweeping.
-
         Returns
         -------
         list
-            A list of all the alternatives for the model config
-            parameters.
+            The run configs currently in this generator
         """
 
-        if type(value) is dict:
-            sweeped_dict = {}
-            for key, sweep_choices in value.items():
-                sweep_parameter_list = []
-
-                # This is the list of sweep parameters. When parsing the Model
-                # Analyzer Config every sweepable parameter will be converted
-                # to a list of values to make the parameter sweeping easier in
-                # here.
-                for sweep_choice in sweep_choices:
-                    sweep_parameter_list += \
-                        self._generate_model_config_combinations(
-                                                    sweep_choice
-                                                    )
-
-                sweeped_dict[key] = sweep_parameter_list
+        return self._run_configs
 
-            # Generate parameter combinations for this field.
-            return self._generate_parameter_combinations(sweeped_dict)
+    def next_config(self):
+        """
+        Returns list of run configs
```

Comment:
Does this return a list of run configs or only the next run config?

Reply:
Ah yes I need to change the docstring